SQLite backend returns integers as strings due to Union type handling bug

## Problem Description

The SQLite storage backend is returning integer values as strings, causing contract test failures in the audit API. Specifically, the `rate_limit` field returns `"100"` (string) instead of `100` (integer).

**Example Test Failure:**
```python
test_get_api_key_success - AssertionError: '100' != 100
```

## Root Cause

The issue is in `campus/storage/tables/backend/sqlite.py` in the `_get_base_type()` function:

1. **Union Type Handling Bug**: The function doesn't handle Union/Optional types (e.g., `int | None`, `Optional[int]`)
2. When a field is typed as `schema.Integer | None`, `_get_base_type()` returns `None` instead of `int`
3. This causes columns to be declared as TEXT instead of INTEGER
4. SQLite returns all values as strings when columns are declared as TEXT

**Current Code:**
```python
def _get_base_type(field_type):
    # Missing Union type handling!
    if field_type in (bool, int, float, str):
        return field_type
    try:
        if issubclass(field_type, int):  # Fails for Union types
            return int
    except TypeError:
        pass
    return None  # Returns None for Union types
```

## Impact

- **API Contract Violations**: Response types don't match API specification
- **Test Failures**: Multiple audit contract tests failing
- **Type Safety Loss**: Applications can't rely on proper Python types from storage

**Affected Tests:**
- `test_get_api_key_success` - rate_limit returns string instead of int
- Potentially other tests using Optional integer fields

## Proposed Solution

### Fix 1: Handle Union Types in `_get_base_type()`

```python
import typing

def _get_base_type(field_type):
    # Handle Union/Optional types
    if typing.get_origin(field_type) is typing.Union:
        args = typing.get_args(field_type)
        for arg in args:
            if arg is not type(None):
                return _get_base_type(arg)
        return None
    
    # Rest of existing logic...
```

### Fix 2: Switch to File-Based Databases (Optional)

Consider using file-based databases instead of `:memory:` for better debugging:

```python
# Current
sqlite3.connect(":memory:")

# Proposed
db_path = os.path.join(tempfile.gettempdir(), f"campus_test_{os.getpid()}.db")
sqlite3.connect(db_path)
```

**Benefits:**
- Data persists for debugging test failures
- Can inspect database state after test runs
- Better alignment with PostgreSQL behavior

## Acceptance Criteria

1. [ ] `_get_base_type()` correctly handles `int | None` and returns `int`
2. [ ] SQLite columns declared as INTEGER instead of TEXT
3. [ ] Contract tests pass with proper integer types in responses
4. [ ] (Optional) File-based databases for test debugging

## Additional Context

The PostgreSQL backend does NOT have this issue because it uses `psycopg2.extras.RealDictCursor` which automatically handles type conversion. This is an SQLite-specific problem.

See: `campus/storage/tables/backend/postgres.py` line 232 for comparison.

--- a/campus/storage/tables/backend/postgres.py
+++ b/campus/storage/tables/backend/postgres.py
@@ -89,12 +89,54 @@ def _validate_field_metadata(field: dataclasses.Field) -> None:
                 )
 
 
+def _get_base_type(field_type):
+    """Get the base Python type for a field type.
+
+    Returns one of: bool, int, float, str (or str as default if not recognized).
+    Handles both built-in types, schema types (which are subclasses),
+    and Union/Optional types (e.g., int | None, Optional[int]).
+    """
+    import types
+    import typing
+
+    # Handle Union/Optional types (e.g., int | None, Optional[int])
+    # Check for both typing.Union (old syntax) and types.UnionType (Python 3.10+ syntax)
+    origin = typing.get_origin(field_type)
+    if origin is typing.Union or origin is types.UnionType:
+        args = typing.get_args(field_type)
+        # Recursively get the base type of the first non-None argument
+        for arg in args:
+            if arg is not type(None):
+                return _get_base_type(arg)
+        return str  # All arguments were None, default to str
+
+    # If it's already a base type, return it directly
+    if field_type in (bool, int, float, str):
+        return field_type
+
+    try:
+        # Check in order: bool, int, float, str (bool is subclass of int)
+        if issubclass(field_type, bool):
+            return bool
+        if issubclass(field_type, int):
+            return int
+        if issubclass(field_type, float):
+            return float
+        if issubclass(field_type, str):
+            return str
+    except TypeError:
+        # issubclass() raises TypeError if field_type is not a class
+        pass
+    return str  # Default to str if not recognized
+
+
 def _field_to_sql_schema(field: dataclasses.Field) -> str:
     """Convert a dataclass field to a SQL column definition."""
     _validate_field_metadata(field)
     field_name = field.name
     field_type = field.type
-    sql_type = _TYPEMAP.get(field_type, "TEXT")
+    base_type = _get_base_type(field_type)
+    sql_type = _TYPEMAP.get(base_type, "TEXT")
     sql_field_constraints = []
 
     if field_name == "__constraints__":

--- a/campus/storage/tables/backend/sqlite.py
+++ b/campus/storage/tables/backend/sqlite.py
@@ -91,8 +91,23 @@ def _get_base_type(field_type):
     """Get the base Python type for a field type.
 
     Returns one of: bool, int, float, str (or None if not a recognized type).
-    Handles both built-in types and schema types (which are subclasses).
+    Handles both built-in types, schema types (which are subclasses),
+    and Union/Optional types (e.g., int | None, Optional[int]).
     """
+    import types
+    import typing
+
+    # Handle Union/Optional types (e.g., int | None, Optional[int])
+    # Check for both typing.Union (old syntax) and types.UnionType (Python 3.10+ syntax)
+    origin = typing.get_origin(field_type)
+    if origin is typing.Union or origin is types.UnionType:
+        args = typing.get_args(field_type)
+        # Recursively get the base type of the first non-None argument
+        for arg in args:
+            if arg is not type(None):
+                return _get_base_type(arg)
+        return None  # All arguments were None
+
     # If it's already a base type, return it directly
     if field_type in (bool, int, float, str):
         return field_type

